test/extended/deployments/deployments.go

```diff
@@ -882,53 +883,85 @@ var _ = g.Describe("deploymentconfigs", func() {
 	})
 
 	g.Describe("with minimum ready seconds set [Conformance]", func() {
+		dc := readDCFixtureOrDie(minReadySecondsFixture)
+		rcName := func(i int) string { return fmt.Sprintf("%s-%d", dc.Name, i) }
 		g.AfterEach(func() {
-			failureTrap(oc, "minreadytest", g.CurrentGinkgoTestDescription().Failed)
+			failureTrap(oc, dc.Name, g.CurrentGinkgoTestDescription().Failed)
 		})
 
 		g.It("should not transition the deployment to Complete before satisfied", func() {
-			_, name, err := createFixture(oc, minReadySecondsFixture)
+			namespace := oc.Namespace()
+			watcher, err := oc.KubeClient().CoreV1().ReplicationControllers(namespace).Watch(metav1.SingleObject(metav1.ObjectMeta{Name: rcName(1), ResourceVersion: ""}))
 			o.Expect(err).NotTo(o.HaveOccurred())
 
-			g.By("verifying the deployment is marked running")
-			o.Expect(waitForLatestCondition(oc, name, deploymentRunTimeout, deploymentRunning)).NotTo(o.HaveOccurred())
+			o.Expect(dc.Spec.Triggers).To(o.BeNil())
+			// FIXME: remove when tests are migrated to the new client
+			// (the old one incorrectly translates nil into an empty array)
+			dc.Spec.Triggers = append(dc.Spec.Triggers, deployapi.DeploymentTriggerPolicy{Type: deployapi.DeploymentTriggerOnConfigChange})
+			dc, err = oc.Client().DeploymentConfigs(namespace).Create(dc)
+
+			g.By("verifying the deployment is created")
+			rcEvent, err := watch.Until(deploymentChangeTimeout, watcher, func(event watch.Event) (bool, error) {
```

Comment:
@kargakis why is it going to flake? If you mean that wait can fail in such short interval I guess retry would be an option. I can't do polling - because I would loose eventual consistency and have a bad resourceVersion and that would cause a flake for sure